voxblox/src/integrator/integrator_utils.cc

```diff
@@ -21,117 +44,27 @@ bool ThreadSafeIndex::getNextIndex(size_t* idx) {
   if (sequential_idx >= number_of_points_) {
     return false;
   } else {
-    *idx = getMixedIndex(sequential_idx);
+    *idx = getNextIndexImpl(sequential_idx);
     return true;
   }
 }
 
 void ThreadSafeIndex::reset() { atomic_idx_.store(0); }
 
-size_t ThreadSafeIndex::getMixedIndex(size_t base_idx) {
-  if (number_of_groups_ * step_size_ <= base_idx) {
-    return base_idx;
+size_t MixedThreadSafeIndex::getNextIndexImpl(size_t sequential_idx) {
+  if (number_of_groups_ * step_size_ <= sequential_idx) {
+    return sequential_idx;
   }
 
-  const size_t group_num = base_idx % number_of_groups_;
-  const size_t position_in_group = base_idx / number_of_groups_;
+  const size_t group_num = sequential_idx % number_of_groups_;
+  const size_t position_in_group = sequential_idx / number_of_groups_;
 
   return group_num * step_size_ + position_in_group;
 }
 
-// Lookup table for the order points in a group should be read in. This is
-// simply a list from 0 to 1023 where each number has had the order of its
-// bits reversed.
-// clang-format off
-const std::array<size_t, ThreadSafeIndex::step_size_>
```

Comment:
Only bit I don't get is how you factored this out?

Reply:
If I remember correctly, it wasn't used anymore. It was just left over from an older version of the threadsafe index.